### PR TITLE
feat: update portal-spec-tests and refactor reading from test files

### PR DESCRIPTION
**Pull request by morph-dev**
### What was wrong?

We were missing test vectors around fork boundaries. These were added to portal-spec-tests in ethereum/portal-spec-tests#49

### How was it fixed?

Updated portal-spec-tests submodule, and simplified/refactored some tests in ehtportal-api crate.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: crates/ethportal-api/src/test_utils/types.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    pub content_key: K,
+    #[serde(rename = "content_value")]
+    pub raw_content_value: RawContentValue,
+}
+
+impl<K: OverlayContentKey> ContentItem<K> {
+    pub fn content_value<V: ContentValue<TContentKey = K>>(&self) -> Result<V, ContentValueError> {
+        V::decode(&self.content_key, &self.raw_content_value)
+    }
+}
```
Comment (KolbyML) [inline crates/ethportal-api/src/test_utils/types.rs:20]: Isn't this type already defined in `crates/utils/src/testing.rs`, shouldn't we only define it once?

File: crates/ethportal-api/src/test_utils/mod.rs [hunk 2]
```diff
@@ -1,18 +1,52 @@
...
+
+/// Reads ssz file from a "portal-spec-tests" submodule
+pub fn read_ssz_portal_spec_tests_file<T: Decode>(path: impl AsRef<Path>) -> anyhow::Result<T> {
+    let bytes = read_binary_portal_spec_tests_file(&path)?;
+    T::from_ssz_bytes(&bytes).map_err(|err| {
+        anyhow!(
+            "Error decoding ssz file: {}. Error: {err:?}",
+            path.as_ref().display()
+        )
+    })
```
Comment (KolbyML) [inline crates/ethportal-api/src/test_utils/mod.rs:51]: Isn't this code already defined in `crates/utils/src/submodules.rs` shouldn't we only define it once?

**Review by KolbyML — COMMENTED** [review]
Most of the PR looks good, I just have questions on the types we already have defined

**Review by KolbyML — APPROVED** [review]
:shipit: looks good. Milos messaged me on discord for context

**Comment by morph-dev** [comment]
> :shipit: looks good. Milos messaged me on discord for context

For context for others, these files and logic is duplicated because we don't want to put it in `ethportal-api` because we publish this crate. And `ethportal-api` can't depend on `crates/utils` because of circular dependency.  So the setup is:
- we have module in `ethportal-api` that is protected by `#[cfg(test)]` that is used only within that crate
- we have `crates/utils` for all other crates
